### DISCUSSION TITLE
Undefined reference to memcpy@GLIBC_2.14

### DISCUSSION THREAD

**Discussion by crb3874** [idaholab/moose#17630] [Q&A Getting Started] [answered]
Hello,

I'm working with a fresh install of conda and MOOSE on an HPC cluster at my university. 

At the end of installation, I'm attempting to run "make -j 4" to compile the tests. When it tries to compile moose/test/src/main.C, it exits with the following errors:

Compiling C++ (in opt mode) (Removed)/moose/test/src/main.C...
/lib64/libresolv.so.2: undefined reference to `__sendmmsg@GLIBC_PRIVATE'

/lib64/libresolv.so.2: undefined reference to `memcpy@GLIBC_2.14' 

/lib64/libresolv.so.2: undefined reference to `__h_errno@GLIBC_PRIVATE'
collect2: error: ld returned 1 exit status

I've looked around - my understanding is that the middle function call here is using version symboling to use the version of memcpy() from GLIBC 2.14. I believe newer versions of GLIBC should be backwards compatible with this feature... I've checked, and the system I'm on is using GLIBC 2.22. Can anyone help me out with this?

**Comment by aeslaughter** [comment, marked as answer]
@milljm Have you seen this before?

**Reply by milljm** [reply]
I have not.

@crb3874 You may want to try and use your system's compilers instead of Conda's. It may yield better results. Can you explain what system you are using? Arch, version, things like that.

**Reply by crb3874** [reply]
Apologies for not following up on this with an update - I ended up starting from scratch by following the instructions at https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html to build MOOSE without using the Conda environment. This worked much better for me, and I am able to compile the tests and run them. However, I am now having a problem (some of the tests failing) that is unrelated to this issue, so I'm going to open a separate discussion for it.

Thank you for your help with this!
